packages/cashier/src/pages/payment-agent/payment-agent-signup/steps/steps-reducer.ts

```diff
@@ -1,17 +1,75 @@
 import { useCallback, useReducer } from 'react';
+import { isEmptyObject } from '@deriv/shared';
 import { ResidenceList } from '@deriv/api-types';
 import { TSelfie } from './selfie/selfie';
+import { Moment } from 'moment';
 
-type TStepsState = {
+//TODO: refactor TSelfie type into TUploadedDocumentType
+
+type TManualPassportValues = {
+    document_id: string;
+    expiry_date: Moment | string;
+    passport: TSelfie | string | null;
+};
+
+type TManualDrivingLicense = {
+    document_id: string;
+    driving_licence_back: TSelfie | string | null;
+    driving_licence_front: TSelfie | string | null;
+    expiry_date: Moment | string;
+};
+
+//TODO: add NIMC type
+
+type TManualIdentityCard = {
+    document_id: string;
+    identity_card_back: TSelfie | string | null;
+    identity_card_front: TSelfie | string | null;
+    expiry_date: Moment | string;
+};
+
+type TManualValues = Partial<TManualPassportValues | TManualDrivingLicense | TManualIdentityCard>;
+
+type TManualErrors = Partial<
+    | Record<keyof TManualPassportValues, string>
+    | Record<keyof TManualDrivingLicense, string>
+    | Record<keyof TManualIdentityCard, string>
+>;
+
+export type TStepsState = {
+    idv_data: {
+        values: {
+            document_number: string;
+            document_type: {
+                example_format: string;
+                id: string;
+                sample_image: string;
+                text: string;
+                value: string;
+            };
+        };
+        errors: { document_number?: string; document_type?: string };
+        country_code?: string;
+    };
+    manual_data: {
+        values: TManualValues;
+        errors: TManualErrors;
+    };
+    selected_manual_document_index: string;
+    is_identity_submission_disabled: boolean;
+    selected_country?: ResidenceList[number];
     selfie: {
         selfie_with_id: TSelfie;
     } | null;
-    selected_country?: ResidenceList[number];
 };
 
 const ACTION_TYPES = {
```

Comment:
@george-usynin-binary I think we can remove this as TS can understand the possible values 🤔 Would you check on this please 🙏🏻